test/HttpTest.cpp

```diff
@@ -54,7 +55,7 @@ TEST(HttpServer, HttpTest) {
       std::vector<ad_utility::JThread> threads;
       for (size_t i = 0; i < 2; ++i) {
         threads.emplace_back([&]() {
-          for (size_t j = 0; j < 5; ++j) {
+          for (size_t j = 0; j < 20; ++j) {
```

Comment:
I would factor this out into a descriptive constant instead of having this magic number just lying around here

Reply:
There already is a comment above the test about spurious failures etc. So it should be clear, that all of those loops can be iterated for longer if you want spurious failures to be less spurious.